### PR TITLE
You've Got Mail

### PR DESCRIPTION
**Pull request by FarkasKBence**
#### Summary
Content "Adds letters, envelopes, greeting cards, gift cards"

#### Purpose of change

It's fluff content.  Survivors should be able to find mail in mailboxes.

#### Describe the solution

Added letters, envelopes, greeting cards, and gift cards.  I figured there's no reason not to have them in-game.  These items serve no gameplay purpose (aside from being firestarters), but it does help with immersion.

Tweaked the `mail` item group to include newly added items.
Greeting cards can now be found in dollar stores and post offices.  Post offices also spawn letters and postcards.

#### Describe alternatives you've considered

Adding more mailing-related items: parcels, postage stamps, utility bills...
Adding more variations to letters and postcards.
Making gift card variants spawn in relevant businesses.

#### Testing

Debug spawned dollar stores and post offices for any mapgen errors.  Debug-spawned an apartment tower to test the `mail` item group.

#### Additional context

`mail` item group:
kép

I apologize for any grammar mistakes.

### CODE REVIEW

File: data/json/items/generic/mail.json [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+    "longest_side": "125 mm",
+    "price": 0,
+    "price_postapoc": 0,
+    "material": [ "plastic" ],
+    "variant_type": "generic",
+    "variants": [
+      {
+        "id": "restaurant_gift_card",
+        "name": { "str": "gift card" },
+        "description": "A gift card for a local restaurant, it covers the cost for an entire meal.",
```
Comment (TheShadowFerret) [inline data/json/items/generic/mail.json:77]:
```suggestion
        "name": { "str": "restaurant gift card" },
        "description": "A gift card for a local restaurant, covering the cost for an entire meal.",
```

File: data/json/items/generic/mail.json [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+      {
+        "id": "clothing_gift_card",
+        "name": { "str": "$20 gift card" },
+        "description": "A $20 gift card, redeemable for purchases at your local clothing store.",
+        "weight": 1
+      },
+      {
+        "id": "barber_gift_card",
+        "name": { "str": "gift card" },
+        "description": "This gift card could be exchanged for a free cut at a local barbershop.",
```
Comment (TheShadowFerret) [inline data/json/items/generic/mail.json:89]:
```suggestion
        "name": { "str": "barbershop gift card" },
        "description": "This gift card could be exchanged for a free cut at a local barbershop.",
```

**Review by TheShadowFerret — COMMENTED** [review]
Since these two variants didn't have unique variant names

**Comment by nexusmrsep** [comment]
It would be nice if letters and cards would be linked to their respective snippets, as a way to smuggle more lore.

**Comment by LovamKicsiGazsii** [comment]
> Since these two variants didn't have unique variant names

Looks great, thanks.

> It would be nice if letters and cards would be linked to their respective snippets, as a way to smuggle more lore.

Similar to newspapers?  That sounds really fun, actually.  I don't have good enough writing skills to do it justice, though.

**Comment by LurchingOtter** [comment]
This is a really good addition to the game - I've always wondered why there are no letters in mailboxes! I think adding snippets for the postcards/letters would be great, too. One of my favourite things about this game is the passive world-building you find in newspapers, flyers, and graffiti. Letters/postcards would be a good addition.

> Similar to newspapers? That sounds really fun, actually. I don't have good enough writing skills to do it justice, though.
Perhaps you could open an issue and ask people for suggestions? If you added even just 5 or 6 snippets each for postcards and letters initially, other people could add more down the line.

**Comment by LovamKicsiGazsii** [comment]
> Perhaps you could open an issue and ask people for suggestions? If you added even just 5 or 6 snippets each for postcards and letters initially, other people could add more down the line.

I'd need to write at least 20 snippets each to avoid repetition.  I'm not planning on adding them in this PR.

Writing letters are much more challenging than I thought.  I have two options: write letters that are suitable for every housing space (rural, one-bedroom, family home, studio apartment), or do separate item groups for each house type for better immersion/sanity.  The latter allows a contributor to write an HOA complaint about one's lawn being too tall without accidentally sending it to a mailbox in an apartment tower/rural house for example.  Or maybe I'm just overcomplicating this.

**Comment by LurchingOtter** [comment]
> I have two options: write letters that are suitable for every housing space (rural, one-bedroom, family home, studio apartment), or do separate item groups for each house type for better immersion/sanity.

That's a good point, I was only really considering a generic letter group that would fit any of those spaces (like a bill or junk mail). You could potentially do a general residential letter group that can be shared between house types, then extend it for each specific residence type. That way you could keep common letters like bills present everywhere, but only have HOA complaints in specific houses. I think that would potentially work well with the mapgen palettes. However it'll be, it makes sense as another PR

> Or maybe I'm just overcomplicating this.

Overcomplicating something in CDDA? Impossible!

**Pull request merged**